### PR TITLE
Add Labels and Annotations to the documentation page.

### LINKED ISSUE #3198: Add labels and annotations to the documentation page
**What would you like to be added**:
As we discusses in https://github.com/kubernetes-sigs/kueue/issues/2858, we have many kueue specific labels and annotations.
So, it would be better to add those to our documentation page.

Labels and Annotations list: https://github.com/kubernetes-sigs/kueue/issues/2858#issuecomment-2379189568

**Why is this needed**:
The documents help batch admins and users understand what happens in Kueue.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Add Labels and Annotations to the documentation page.

#### Which issue(s) this PR fixes:

Fixes #3198

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+This page serves as a reference for all labels and annotations in Kueue.
+
+
+### kueue.x-k8s.io/queue-name
+
+Type: Label
+
+Example: `kueue.x-k8s.io/queue-name: "my-local-queue"`
+
+Used on: Jobs.
```
Comment (kannon92) [inline site/content/en/docs/reference/labels-and-annotations.md:16]: We probably shouldn’t use Jobs here…. I don’t know what we should use here. Job to me means a batch job but this label is required for any Kueue supported job (ie Job, JobSet, Ray, Pod, Deployment, etc)

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+---
+title: "Labels and Annotations"
+linkTitle: "Labels and Annotations"
+date: 2024-10-09
+---
+
+This page serves as a reference for all labels and annotations in Kueue.
```
Comment (mwielgus) [inline site/content/en/docs/reference/labels-and-annotations.md:7]: Can we have these labels/annotations sorted?

Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md:7]: I sorted them by most used. Which order do you prefer?

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 1 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md:16]: Yeah, I encountered the same question.

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 2 above)
Comment (tenzen-y) [reply site/content/en/docs/reference/labels-and-annotations.md:7]: In general, users do not have a context which labels and annotations are mostly used.
So, alphabetical order would be better, IMO.

Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md:7]: > In general, users do not have a context which labels and annotations are mostly used. So, alphabetical order would be better, IMO.

Make sense.

Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md:7]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 1 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Maybe "Kueue-managed Jobs."?

Comment (tenzen-y) [reply site/content/en/docs/reference/labels-and-annotations.md]: This looks great.

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/is-group-workload
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/is-group-workload: "true"`
+
+Used on: [Workload](/docs/concepts/workload/).
+
+The label key indicates that this workload is used for a group of Pods.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:18]:
```suggestion
The annotation key indicates that this workload is used for a group of Pods.
```

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/job-completions-equal-parallelism
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/job-completions-equal-parallelism: "true"`
+
+Used on: [batch/Job](/docs/tasks/run/jobs/).
+
+The label key is used to keep `completions` and `parallelism` in sync
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:29]:
```suggestion
The annotation key is used to keep `completions` and `parallelism` in sync
```

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/job-min-parallelism
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/job-min-parallelism: "5"`
+
+Used on: [batch/Job](/docs/tasks/run/jobs/).
+
+The label key indicates the minimum `parallelism` acceptable for the job in the case of partial admission.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:40]:
```suggestion
The annotation key indicates the minimum `parallelism` acceptable for the job in the case of partial admission.
```

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+The label key in the workload resource holds the UID of the owner job.
+
+
+### kueue.x-k8s.io/managed
+
+Type: Label
+
+Example: `kueue.x-k8s.io/managed: "true"`
+
+Used on: Resources managed by Kueue.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:60]: Is this correct? AFAIK, this label is used only in the Pod integration, right?
Could you re-check if this label is used in other resources?

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/multikueue-origin
+
+Type: Label
+
+Example: `kueue.x-k8s.io/multikueue-origin: "true"`
+
+Used on: [MultiKueue](/docs/concepts/multikueue/).
+
+The label key is used to track the creator of MultiKueue remote objects.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:73]: Could you explain what is "remote objects"? I am thinking if we should say "Objects in Worker Cluster".

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/pod-group-total-count
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/pod-group-total-count: "2"`
+
+Used on: [Plain Pods](/docs/tasks/run/plain_pods/).
+
+The label key is used to indicate how many Pods to expect in the group.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:95]:
```suggestion
The annotation key is used to indicate how many Pods to expect in the group.
```

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/role-hash
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/role-hash: "b54683bb"`
+
+Used on: [Plain Pods](/docs/tasks/run/plain_pods/).
+
+The label key is used as the name for a Workload podSet.
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:169]:
```suggestion
The annotation key is used as the name for a Workload podSet.
```

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+### kueue.x-k8s.io/retriable-in-group
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/retriable-in-group: "false"`
+
+Used on: [Plain Pods](/docs/tasks/run/plain_pods/).
+
+The label key is used to finalize the group if at least one terminated Pod (either Failed or Succeeded)
```
Comment (tenzen-y) [inline site/content/en/docs/reference/labels-and-annotations.md:157]:
```suggestion
The annotation key is used to finalize the group if at least one terminated Pod (either Failed or Succeeded)
```

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 4 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 3 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 8 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 9 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 10 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 6 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md:60]: You right, my mistake. Thanks!

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 3 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Thanks!

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 7 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 6 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

**Pull request merged**